geometric/tests/test_prepare.py

```diff
@@ -94,12 +94,40 @@ def test_parse_constraints(localizer):
     np.testing.assert_almost_equal(vals[13][24], 0.5/0.529177210903)
     np.testing.assert_almost_equal(vals[14][25], 0.0)
 
-def test_get_molecule_engine_pdb():
-    for engine, inputf in zip(['terachem', 'qchem', 'molpro', 'psi4', 'gaussian'], ['water12.tcin', 'water12.qcin', 'water12.mol', 'water12.psi4in', 'water12.gjf']):
-    # for engine, inputf in zip(['terachem', 'qchem'], ['water12.tcin', 'water12.qcin']):
-        molecule, engine = geometric.prepare.get_molecule_engine(input=os.path.join(datad, inputf), engine=engine,
-                                                                 pdb=os.path.join(datad, 'water12.pdb'), 
-                                                                 coords=os.path.join(datad, 'water12.mdcrd'))
-        assert molecule.resid[-1] == 12
-        np.testing.assert_almost_equal(molecule.xyzs[0][-1,2], 1.288)
-        assert engine.detect_dft() == False
+
+@addons.using_terachem
+def test_get_molecule_engine_pdb__terrachem():
+    get_molecule_engine_pdb("terachem", "water12.tcin")
+
+
+def test_get_molecule_engine_pdb__psi4():
```

Comment:
Is the double underscore after `pdb` intentional? (Same question for other instances below).

Thanks for adding `get_molecule_engine_pdb` and the tests using the other engines/calculators!

Reply:
Yes, having broken up a test into engine-specific versions, this separates the name of the test from the engine. Most likely we could do this with pytest's parametrisation as well, though I am not sure how to add the decorators and likely it is not worth the hassle for now.